### ISSUE TITLE
Feature Request:  First class Ownership field

### ISSUE THREAD

**Issue by twisted3motions** [grokability/snipe-it#7662]
**Server (please complete the following information):**
 - Snipe-IT Version 4.8
 - OS: Ubuntu 18.04
 - Web Server: Apache
 - PHP Version 7.3

**Is your feature request related to a problem? Please describe.**
Probably the biggest need I have is an "owner" field.  

At the university I work at each machine is managed by an IT group which fits into a company nicely.  For example.... You may have companies like College of Engineering, School of Business, Department of Administration.  Each of those would have their own admins and need permissions segregated.  Company maps well to tracking the custodian or manager of the asset.

However, beneath that... each machine is owned by a particular department, lab, or other group.  This needs to be tracked.  There are many use cases I won't bore you with.  Under my IT department alone there are 30 research labs with their own lists of equipment we inventory for them.

Anyways.  Just because the asset is checked out to John or managed by Engineering IT doesn't mean that it doesn't belong to the Fusion Lab.  This info is very important, but we are lucky to know the list of possible owners before hand.

I'm surprised other than company (which seems uniquely fit for situations where there are admins of different groups), there is no default field indicating asset ownership.  In a private institution I suppose everything would belong to one identity but here we have individual labs that use their grant moneys to purchase equipment only for their use.

**Describe the solution you'd like**

Using a custom field stinks as I can't change its position to just under the company field.  By default its also one of the last columns in the list view.  The owner field is probably one of the most important, used, and viewed fields. 

With a custom field I also don't get a nice effect of clicking on a blue link and being shown info about that owner and all the assets they own.  

I would love an owner field that behaves like the company field and that is a first class citizen in the user interface.

**Comment by twisted3motions** [comment]
Departments field unfortunately is just a container of users and isn't applied to assets.

**Comment by Sxderp** [comment]
I had requested something similar. https://github.com/snipe/snipe-it/issues/7544
Not much traction.
=-/

**Comment by Sxderp** [comment]
yes

**Comment by pbrink231** [comment]
got around to checking this out.  made department apart of an asset.

https://github.com/pbrink231/snipe-it/tree/feature-58-add-department-to-asset-accessory

pull request:
#7843 

Will make a pull request when I get a chance.

<blockquote><img src="https://avatars1.githubusercontent.com/u/8224560?s=400&v=4" width="48" align="right"><div><img src="https://github.githubassets.com/favicon.ico" height="14"> GitHub</div><div><strong><a href="https://github.com/pbrink231/snipe-it">pbrink231/snipe-it</a></strong></div><div>A free open source IT asset/license management system - pbrink231/snipe-it</div></blockquote>

**Comment by Sxderp** [comment]
I'm not sure just adding department to an asset solves the overall issue.

**Comment by pbrink231** [comment]
Our situation sounds similar to yours.

We have a factory which actually has assets from several companies.  We also have departments or areas you could say that purchase their own items and manage them.

A couple scenarios
IT department manages all computers in the factory.
Cutting department manages their plotting machines, cutting tables, cutting machines
Sub Printing area has a their own consumables they manage.
several other departments with a mix of assets managed by them and managed by IT or others.

@Sxderp 
The department field is already connected with a company and users.  So future could add permissions down to an asset level based on user and department.
In your scenario would a department, lab, group not just be a department in the company that you could set as the manager of the asset?
Can you describe how that does not solve the issue and how the owner field would work different in your scenario?

Could also set a flag on department like isOwnerable or something so only those departments can be used.

**Comment by Sxderp** [comment]
Now that I've taken the time to review, I think using department would be sufficient.

**Comment by Sxderp** [comment]
yes

**Comment by Sxderp** [comment]
yes

**Comment by twisted3motions** [comment]
Yep.

**Comment by Sxderp** [comment]
yes

**Comment by Sxderp** [comment]
yes

**Comment by FreeMinded** [comment]
This feature would be very useful for our use case. We (will) have the assets of our customers (nicely mapping to companies) in SnipeIT while the assets itself can either belong a) to the customer, b) to us as service provider or c) to a third party in case of rental/leasing.

My favorite solution would be an Ownership field as proposed by @twisted3motions. Technically it should be a "copy" of the company field with dropdown from the same list of companies.

Alternatively the _custom field form element_ **List Box** could be extended in such a way that an existing data set (in this case companies) could be used as field value instead of having to add them to the text box.

It works with custom fields the way they today are but a first class Ownership field would make it so much better.

The department functionality does not work for our use case.

**Comment by snipe** [comment]
I feel like this is probably solved via the "user can belong to multiple companies" and also the floater mode solutions we've introduced over the past months.

**Comment by FreeMinded** [comment]
@snipe I disagree on this. The feature request has nothing to do with the user but with the ownership of the asset. My comment from above is still valid. https://github.com/grokability/snipe-it/issues/7662#issuecomment-1386137395
